src/arrayMethodSort.js

```diff
@@ -4,8 +4,58 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  // Define the custom sort2 method on the Array prototype
+  [].__proto__.sort2 = function (compareFunction) {
+    const arr = this;
+    const len = arr.length;
+
+    const defaultCompare = (a, b) => {
+      const aStr = String(a);
+      const bStr = String(b);
+
+      if (aStr < bStr) {
+        return -1;
+      }
+
+      if (aStr > bStr) {
+        return 1;
+      }
+
+      return 0;
+    };
+
+    const compare =
+      typeof compareFunction === 'function' ? compareFunction : defaultCompare;
+
+    /**
+     * In-place Quicksort implementation
+     * @param {number} start
+     * @param {number} end
+     */
+    function quickSort(start, end) {
+      if (start >= end) {
+        return;
+      }
+
+      const pivot = arr[end];
+      let i = start;
+
+      for (let j = start; j < end; j++) {
+        if (compare(arr[j], pivot) < 0) {
+          [arr[i], arr[j]] = [arr[j], arr[i]];
```

Comment:
You are using array destructuring to swap elements in the array, which is a very clean and efficient way to do it. However, it's worth noting that this syntax can be confusing for beginners, so a comment explaining what it does could be helpful.